### DISCUSSION TITLE
When will the fix for the issue connecting to DS over SSL using LdapSessionOptions on Linux be released?

### DISCUSSION THREAD

**Discussion by rafeeqg** [dotnet/runtime#56771] [Q&A] [answered]
On Linux, we're having issues connecting to OID using LdapSessionOptions via SSL. I've been keeping an eye on MR #52904, but that does not appear to be included in the current runtime version 5.0.3. Could you please tell us when this fix will be ready and in which version?

**Code**
```
public string Connect()
        {
            var identifier = new LdapDirectoryIdentifier("xxx.xxxx.com", _securePortNumber, true, false);
            var connection = new LdapConnection(identifier)
            {
                AuthType = AuthType.Basic
            };
            connection.SessionOptions.ProtocolVersion = 3;

            var credential = new NetworkCredential("xxxx", "xxxx");
            connection.Credential = credential;

            try
            {
                connection.SessionOptions.SecureSocketLayer = true;
                connection.SessionOptions.VerifyServerCertificate = new VerifyServerCertificateCallback((connection, certificate) => true);
                connection.Bind();
            }
            catch (LdapException ex)
            {
                Console.WriteLine(ex);
            }
            return "Success!";
        }
```

**Runtime Version Details**

dotnet --version

> 5.0.302

dotnet --list-runtimes

> Microsoft.AspNetCore.All 2.1.28 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
> Microsoft.AspNetCore.App 2.1.28 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
> Microsoft.AspNetCore.App 3.1.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
> Microsoft.AspNetCore.App 5.0.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
> Microsoft.NETCore.App 2.1.28 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
> Microsoft.NETCore.App 3.1.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
> Microsoft.NETCore.App 5.0.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
> Microsoft.WindowsDesktop.App 3.1.17 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
> Microsoft.WindowsDesktop.App 5.0.8 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

**Exception Details**

> fail: Microsoft.AspNetCore.Server.Kestrel[13]
>       Connection id "0HMAMBG2KFI85", Request id "0HMAMBG2KFI85:00000002": An unhandled exception was thrown by the application.
>       System.PlatformNotSupportedException: Operation is not supported on this platform.
>          at System.DirectoryServices.Protocols.LdapSessionOptions.set_SecureSocketLayer(Boolean value)
>          at DotNet5LDAP.Controllers.ConnectController.Post() in /src/Controllers/ConnectController.cs:line 47
>          at lambda_method1(Closure , Object , Object[] )
>          at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.SyncObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
>          at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeActionMethodAsync()
>          at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
>          at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeNextActionFilterAsync()
>       --- End of stack trace from previous location ---
>          at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
>          at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
>          at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
>       --- End of stack trace from previous location ---
>          at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
>          at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
>          at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
>          at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
>          at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)

**Comment by bartonjs** [comment, marked as answer]
It won't be in the 5.0.x packages at all, it's a feature for 6.0.0.  I believe it should be present in the 6.0.0-preview.6 version of the package.

**Reply by rafeeqg** [reply]
Thanks @bartonjs. I'll mark this as answer
